Packages/doc/databrowser_paplot.rst

```diff
@@ -0,0 +1,223 @@
+.. _db_paplot:
+
+Pulse Average (PA) plot
+=======================
+
+When the stimulus set contains multiple pulses the pulse average plot allows to
+extract these single pulses to overlay, and average and visually inspect these.
+
+Although the stimulus set will contain :ref:`wavebuilder_pulsetrain` epochs in
+most cases, the pulse extraction algorithm works with all epoch types and
+special measurement modes like :ref:`daephys_dDAQ` or :ref:`daephys_oodDAQ`.
+
+The dDAQ slider from the :ref:`db_settings` tab is respected as is the :ref:`db_channelselection`.
+
+For instructional purposes assume you have acquired a pulse train with oodDAQ on 4 different headstages.
+This means you have 4 active headstages with each a DA and AD channel. We
+only visualize the AD data in the PA plot.
+
+The PA plot will then have 16 = 4 x 4 PA sets (either in one big graph or in multiple graphs).
+
+Each set will have one source of pulse starting times (called "region")
+and one source of the visualized data. The pulse starting times are always
+extracted from the DA channel of the region.
+
+That means the top left set is from the first region and the first
+active headstage, the one to the right is from the second region and the
+first active headstage. Or to put it differently, regions are the columns.
+
+The following two figures, :ref:`Figure Pulse Average Trace plot` and
+:ref:`Figure Pulse Average Image plot`, were created with multiple sweeps
+overlayed, see :ref:`db_overlaysweeps`, and time alignment and zeroing turned on.
+
+.. _Figure Pulse Average Trace plot:
+
+Trace plot
+""""""""""
+
+.. figure:: PulseAverage_traces.png
+   :align: center
+   :width: 75%
+
+   Trace plot
+
+The trace plot is a display of all single pulses from a PA set as traces. The
+scale bars, shown in black, give an impression of the data range. The trace
+colors coincide with the trace colors used in the databrowser and denote the
+headstage of the data, see also :ref:`Figure Relevant Colors`.
+
+.. _Figure Pulse Average Image plot:
+
+Image plot
+""""""""""
+
+.. figure:: PulseAverage_images.png
+   :scale: 50%
+   :figwidth: 70%
+   :align: left
+
+.. figure:: PulseAverage_images_colorscales.png
+   :height: 870px
+   :align: right
+
+.. raw:: html
+
+   <p>&nbsp;&nbsp;&nbsp;<br></p>
+
+This graph is an alternative display to the :ref:`Figure Pulse Average Trace
+plot`. Its advantage is a much faster updates for displaying many overlayed
+sweeps. Each set image has the average and the deconvolution data at the
+bottom. When these are not shown a blank space is left empty.
+On top of these the single pulse data is shown. Each data is displayed as rows
+(horizontally), and the image is filled from bottom to top depending on the
+:ref:`db_paplot_image_sortorder`.
```

Comment:
The image graph supplements the trace plot :ref:`Figure Pulse Average Trace plot`. It renders more quickly than the trace plot, especially with many (overlayed) sweeps. Each (horizontal) line of the image plot corresponds to a pulse region (or single time-series) and encodes the voltage or current in color (user-defined color mapping). Deconvolution and average lines (extra wide) are at the bottom of each set image. Image-space is left blank when data is not shown. The image is filled from bottom to top depending on the :ref:`db_paplot_image_sortorder`.

Reply:
@t-b can we reverse the image filling to top to bottom?

Reply:
@timjarsky I've created #645.

Reply:
I also used `pulse` instead of `pulse region`, as we are already using region meaning one column of the plot and I though this to be confusing.